events/code_review_events/workflow.py

```diff
@@ -216,9 +257,140 @@ def publish_results(self, payload):
 
         return True
 
-    def parse_pulse(self, payload):
+    async def got_bugbug_test_select_end(self, payload):
```

Comment:
I would put that code in a separate module/class/function to be able to explicitely try/catch it, and easily add unit tests.
That makes workflow.py really big :sob:

Reply:
Done.